### PR TITLE
Add Network Static Route/Gateway support

### PR DESCRIPTION
**Pull request by raviteja-b**
This commit enables static route configuration on EthernetInterface Implements CreateStaticRoute method which creates a new d-bus object with StaticRoute interface.

Tested By:
Run StaticRoute D-bus method and verified D-bus object and configuration Delete StaticRoute object

Change-Id: I3fbc6f85ede00b6c1949a0ac85f501037a69c831

### CODE REVIEW

**Comment by raviteja-b** [comment]
upstream commit: https://gerrit.openbmc.org/c/openbmc/phosphor-networkd/+/63033

File: test/test_ethernet_interface.cpp [hunk 1]
```diff
@@ -281,5 +281,55 @@ TEST_F(TestEthernetInterface, AddMultipleIPv6StaticGateways)
                                      Key(std::string("2004:903:15f:325::1"))));
 }
 
+TEST_F(TestEthernetInterface, AddStaticRoute)
```
Comment (sunharis) [inline test/test_ethernet_interface.cpp:284]: duplicates. please remove

**Review by sunharis — CHANGES_REQUESTED** [review]

File: test/test_ethernet_interface.cpp (hunk 1 above)
Comment (raviteja-b) [reply test/test_ethernet_interface.cpp]: done

File: src/ethernet_interface.cpp [hunk 2]
```diff
@@ -682,6 +683,58 @@ void EthernetInterface::loadNameServers(const config::Parser& config)
...
+void EthernetInterface::loadStaticRoutes(const config::Parser& config)
+{
+    std::vector<std::string> destinations =
+        config.map.getValueStrings("Route", "Destination");
+    std::vector<std::string> gateways =
+        config.map.getValueStrings("Route", "Gateway");
+    for (uint8_t i = 0; i < destinations.size() && i < gateways.size(); i++)
+    {
+        size_t pos = destinations[i].find("/");
+        std::string dest = destinations[i].substr(0, pos);
```
Comment (asmithakarun) [inline src/ethernet_interface.cpp:695]: Should we validate the IP and the prefix here?

Comment (raviteja-b) [reply src/ethernet_interface.cpp:695]: validation will done here
```
  try
        {
            ifaddr = {addr, prefix};
        }
        catch (const std::exception& e)
        {
            auto msg = fmt::format("Invalid static route {}\n", e.what());
            log<level::ERR>(msg.c_str());
        }
```

Comment (asmithakarun) [reply src/ethernet_interface.cpp:695]: Okay

**Review by asmithakarun — APPROVED** [review]

**Pull request merged**